src/processTargets/modifiers/scopeTypeStages/RegexStage.ts

```diff
@@ -70,7 +65,7 @@ class RegexStage implements ModifierStage {
     contentRange: Range
   ): ScopeTypeTarget {
     return new ScopeTypeTarget({
-      scopeTypeType: this.modifier.scopeType.type,
+      scopeTypeType: this.modifier.scopeType.type as SimpleScopeTypeType,
```

Comment:
Hmm.  Not in love with the cast.  Why do we need `scopeTypeType` on `ScopeTypeTarget` anyway?

Reply:
Well we use it for creating continuous ranges and default delimiters. Not really something we can just remove without consequence.

Reply:
Ah right yeah default delimiters.

I still don't love the cast here. It's not actually a simple scope type type in the case of a custom regex right? I think a cast is ok when the compiler just isn't smart enough to do inference. In this case we're actually lying to the compiler right?

Reply:
Yeah somewhat

Reply:
This is the only real showstopper for me.  There are a few ways to handle it.  Any preference?

Reply:
Not really

Reply:
Maybe just pass in the `scopeType` itself instead of  the `scopeTypeType`?

Alternately don't pass in the scopeTypeType at all and move the switch statement that uses it into the syntactic scope stage

Reply:
I guess we could do that. 


We use it in multiple places so that would not be an easy fix.

Reply:
Do you have a simpler solution? I don't think that lying to the type system is a good idea, so we need to figure something out

Reply:
I think the simplest solution is prob to make the scopeTypeType arg be nullable, ie `ScopeTypeType | null`. And then just pass in `null` here. Maybe not the most precise but at least it's not a lie 😅

Reply:
If we're not going to rely on anything it has to do with the scope type maybe we should use a different target instead?

Reply:
Fine by me